backend/src/main/kotlin/no/nav/tilbakekreving/burdeforstatt/service/MQService.kt

```diff
@@ -0,0 +1,36 @@
+package no.nav.tilbakekreving.burdeforstatt.service
+
+import jakarta.jms.*
+import no.nav.tilbakekreving.burdeforstatt.config.MqConfig
+import no.nav.tilbakekreving.burdeforstatt.kontrakter.Ytelsestype
+import no.nav.tilbakekreving.burdeforstatt.util.Marshaller
+import no.nav.tilbakekreving.kravgrunnlag.detalj.v1.DetaljertKravgrunnlagMelding
+import org.slf4j.LoggerFactory
+
+class MQService (private val mqConfig: MqConfig) {
+    private val log = LoggerFactory.getLogger(this::class.java)
+
+    fun sendMessage(detaljertKravgrunnlagMelding: DetaljertKravgrunnlagMelding, eksternFagsakId: String, ytelsestype: Ytelsestype) {
+        try {
+            val connection = mqConfig.createConnection()
+            val session: Session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)
+            val queue: Queue = session.createQueue(mqConfig.queue)
+            val producer: MessageProducer = session.createProducer(queue)
+
+            val dtoXml = Marshaller.marshall(detaljertKravgrunnlagMelding)
+
+            val message: TextMessage = session.createTextMessage(dtoXml)
+
+            producer.send(message)
+            connection.close()
+        } catch (e: JMSException) {
+            log.warn("Kunne ikke sende kravgrunnlag med id {} til MQ", detaljertKravgrunnlagMelding.detaljertKravgrunnlag?.kravgrunnlagId)
+            log.error("JMSException occurred: {}", e.message)
+            e.printStackTrace()
```

Comment:
Vil helst ikke kalle e.printStackTrace, bedre å sende exception til loggeren i stedet for kun meldingen. Vi burde nok også passe på at feilen ikke bare blir spist i catchen, men faktisk ender opp hos brukeren